src/main/java/com/github/pires/hazelcast/HazelcastDiscoveryController.java

```diff
@@ -119,8 +120,17 @@ public void run(String... args) {
             if (endpoints != null) {
                 if (endpoints.subsets != null && !endpoints.subsets.isEmpty()) {
                     endpoints.subsets.forEach(subset -> {
-                        subset.addresses.forEach(
+                    	if (subset.addresses != null && subset.addresses.isEmpty()) {
```

Comment:
You meant `!subset.addresses.isEmpty()`, otherwise this will never run if there are ready addresses.